src/ServiceClient.js

```diff
@@ -0,0 +1,157 @@
+import grpc, { InterceptingCall } from "grpc";
+import url from "url";
+import { BigNumber } from 'bignumber.js';
+import { find, map } from 'lodash';
+import paymentChannelStateServices from './payment_channel_state_service_grpc_pb';
+
+export default class ServiceClient {
+  constructor(sdk, metadata, group, ServiceStub, paymentChannelManagementStrategy) {
+    this._sdk = sdk;
+    this._web3 = this._sdk.web3;
+    this._account = this._sdk.account;
+    this._mpeContract = this._sdk.mpeContract;
+    this._metadata = metadata;
+    this._group = group;
+    this._grpcStub = this._generateGrpcStub(ServiceStub, paymentChannelManagementStrategy);
+    this._paymentChannelStateServiceClient = this._generatePaymentChannelStateServiceClient();
+    this._paymentChannels = [];
+  }
+
+  get stub() {
+    return this._grpcStub;
+  }
+
+  get group() {
+    return this._group;
+  }
+
+  get groupIdInBytes() {
+    return Buffer.from(this.group.group_id, 'base64');
+  }
+
+  get paymentChannels() {
+    return this._paymentChannels;
+  }
+
+  get metadata() {
+    return this._metadata;
+  }
+
+  get paymentAddress() {
+    return this.group.payment_address;
+  }
+
+  get paymentChannelStateServiceClient() {
```

Comment:
I would made all methods which are not part of API private

Reply:
[MPEContract](https://github.com/singnet/snet-sdk-js/pull/1/files#diff-0e7c20456d55f6ddea0245e78ecb692dR97) needs it so that it can create a `PaymentChannel` with it.

In general all the methods which are used outside the respective classes have been made public, either within the SDK or outside. This way refactoring becomes more easy even for code within SDK code.

Reply:
Am I right that int this implementation `MPEContract` uses channel state received from daemon? If so then I would say `MPEContract` is misleading name as it supposes that the class is kind of Ethereum contract wrapper.

Reply:
Well you are right in the sense that `MPEContract` is mostly an Ethereum contract wrapper. But however it also adds some helper methods to get all the open channels between a service and an account. Now we could move it out of `MPEContract` but I felt that this is a good place for it to be in. If this means renaming the class name, I think that makes sense. Any suggestions? How about `MPEOperations`?

Reply:
I am not sure we need `MPEContract` as part of public API. At the moment there is also `Account` which has the same methods in API and used by `MPEContract` in fact. What is the reason of having them both?

Reply:
The thing which confuses me is that `Account` have many responsibilities:
- it signs transactions
- it sends transactions
- it manipulates balances

I would expect that signing functionality is isolated in separate class which manages private key, in our diagrams it is one of `Identity` implementations. Sending transaction functionality stays in `MPEContract` wrapper as it should simplify contract calling. And manipulating balances is `Account` responsibility and it uses `Identity` and `MPEContract` to do this.

Reply:
Agree. Identity will soon follow in the next version when we start support more than one Identity types. For this release we are just sticking to `PK`.